modules/web/js/ballerina/views/ballerina-file-editor.js

```diff
@@ -329,20 +330,23 @@ define(['lodash', 'jquery', 'log', './ballerina-view', './service-definition-vie
                 // adding current package to the tool palette with functions, connectors, actions etc. of the current package
                 this.addCurrentPackageToToolPalette();
             }
-            //adding default packages TODO : this needs to be rendered by referring to imports in the model
-            var httpPackage = BallerinaEnvironment.searchPackage("ballerina.net.http");
-            this.toolPalette.getItemProvider().addImport(httpPackage[0]);
-
-            // render tool palette
-            this.toolPalette.render();
 
+            var importDeclarations = [];
             if(!this._parseFailed){
                 // Creating the constants view.
                 this._createConstantDefinitionsView(this._$canvasContainer);
                 this._model.accept(this);
+                importDeclarations = this._model.getImportDeclarations();
             }
 
-            var self = this;
+            // add current imported packages to tool pallet
+            _.forEach(importDeclarations, function (importDeclaration) {
+                var package = BallerinaEnvironment.searchPackage(importDeclaration.getPackageName());
+                self.toolPalette.getItemProvider().addImport(package[0]);
```

Comment:
Yeah, it could have been null. But now the tool pallet is rendered declaratively using react. So all the imports in the ast root node is added rendered in the tool pallet. The problem does not occur anymore.